HybridCar.cpp

```diff
@@ -1,15 +1,29 @@
 #include "HybridCar.hpp"
 #include <iostream>
 
-HybridCar::HybridCar(PetrolEngine* petrolEng, ElectricEngine* electricEng)
-    : PetrolCar(petrolEng)
-    , ElectricCar(electricEng)
-{
+HybridCar::HybridCar(Engine* elEngine, PetrolEngine* petrolEng)
+    : ElectricCar(elEngine), engine_(petrolEng) {
     std::cout << __FUNCTION__ << std::endl;
 }
 
-HybridCar::~HybridCar()       { std::cout << __FUNCTION__ << std::endl; }
+HybridCar::~HybridCar()       {std::cout << __FUNCTION__ << std::endl; }
 void HybridCar::feed() {
-    PetrolCar::feed();
     ElectricCar::feed();
- }
\ No newline at end of file
+ }
+
+void HybridCar::changeEngine(Engine *engine) {
+
+    if (velocity != 0)
+        throw std::logic_error( "Car is moving now! Engine exchange impossible ");
+    
+    auto* engineE = dynamic_cast<ElectricEngine*>(engine);
+    if (engineE != nullptr) {
+        ElectricCar::changeEngine(engine);
+        delete (engineE);
+    }
+
+    auto* engineP = dynamic_cast<PetrolEngine*>(engine);
+    if (engineP != nullptr) {
+        throw std::runtime_error("Change of petrol engine failed");
```

Comment:
Nigdzie nie przypisujemy tego silnika jeśli uda się konwersja tylko rzucamy wyjątkiem. Chyba inaczej to miało działać.